### ISSUE TITLE
typecheck settings is not respected

### ISSUE THREAD

**Issue by Eliav2** [vitest-dev/vscode#367]
### Describe the bug

typecheck settings is not respected

![tmp](https://github.com/vitest-dev/vscode/assets/47307889/d82ed87b-adde-46f7-ae34-68c866fb445e)

in the gif you can see:
- running `npx vitest` correctly check typing.
- running the test from the IDE extension does not catch type error, even though this file only checks for typing only
- this extension should respect filed `*.test-d.ts` for type tests only tests.

### Reproduction

first of all, awesome extension!

to reproduce:
```
git clone https://github.com/Eliav2/express-typed
// or, checkout exact commit (in case the repo would be changed)
git clone --bare https://github.com/Eliav2/express-typed.git
cd express-typed.git
git checkout 6aec67bfe1c849edfac0a85d41a1df0faa60cb0d
```

now run `npx vitest` at root directory, or run tests from the IDE.

#### OR small example

from [vitest docs](https://vitest.dev/guide/testing-types.html#testing-types) - [Play Online](https://stackblitz.com/fork/github/vitest-dev/vitest/tree/main/examples/typecheck?initialPath=__vitest__/)

This extension would not correctly catch [`type.test-d.ts`](https://stackblitz.com/edit/vitest-dev-vitest-u3faap?file=test%2Ftype.test-d.ts&initialPath=__vitest__/) type errors.

### Output

```shell
.
```

### Version

v0.8.6

### Validations

- [X] Check that you are using the latest version of the extension
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vscode/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vscode/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
Vitest shows the errors as unhandled source errors. To see the error alongside the test, you need your test file to match the `typecheck.include` matcher.

But there is indeed a bug where Vitest extension doesn't print any unhandled errors at all.

**Comment by Eliav2** [comment]
Type check is enabled and handled correctly in my example by vitest himself. The extension doesn't catch these error because probably it's running with its own configurations and doesn't respect the vitest config used in the workspace

**Comment by sheremet-va** [comment]
No, the extension doesn't modify your configuration, it just doesn't report them because there is no way to assign them to a test. If you read the output of Vitest in the terminal in your gif, you can see that it says "Source Errors", and not test errors. In the same gif you can see "Type Errors: no errors".

**Comment by Eliav2** [comment]
@sheremet-va notice, not "source errors" but "Type errors", which is supported(experimentaly) in vitest.
I can see you already linked a pull request with a fix  👍💯

**Comment by sheremet-va** [comment]
> not "source errors" but "Type errors",

![image](https://github.com/vitest-dev/vscode/assets/16173870/1036d104-3981-45d6-b0fd-860664806ba1)

**Comment by Eliav2** [comment]
As long as your pr would fix expectTypeOf and assertType assertions on *.test-d.ts files in the vscode extension we're on the same page
